src/files/views.py

```diff
@@ -66,6 +70,38 @@ def get_object(self, queryset: QuerySet[BaseFile] | None = None) -> BaseFile:
         return basefile  # type: ignore[no-any-return]
 
 
+class FileDeleteView(DeleteView):  # type: ignore[type-arg]
+    """File delete view. Delete a single file."""
+
+    template_name = "delete.html"
+    model = BaseFile
+    success_url = reverse_lazy("files:file_list")
+
+    def get_object(self, queryset: QuerySet[BaseFile] | None = None) -> BaseFile:
+        """Check permissions before returning the file."""
+        basefile = super().get_object(queryset=queryset)
+        if not self.request.user.has_perm("files.delete_basefile", basefile):
+            # file is not PUBLISHED, and the current user does not have permissions to view this file
+            raise PermissionDenied
+        return basefile  # type: ignore[no-any-return]
```

Comment:
deleteview should softdelete the file and return 204, not return the file

Reply:
Fixed, but it redirect back to the detail page instead of a 204.